src/TorSharp/ToolUtility.cs

```diff
@@ -232,5 +282,38 @@ public static ToolSettings GetTorToolSettings(TorSharpSettings settings)
                 .OrderByDescending(t => t.Version)
                 .FirstOrDefault();
         }
+
+        public static bool TryFindToolInSystem(TorSharpSettings settings, ToolSettings toolSettings, out Tool tool)
+        {
+            tool = null;
+            if (settings.OSPlatform != TorSharpOSPlatform.Linux) // For windows search exe name in PATH variable?
+                return false;
+
+            // In linux most binaries exists in any bin directory, e.g /bin/ /sbin/ /usr/bin/
+            var toolPath = WhereIsUtility.WhereIs(toolSettings.TryFindExecutableName);
```

Comment:
I'm curious why `whereis` is used instead of `which`. Isn't the goal here just to find some system-provided executable? I think one benefit of using `which` is that it only returns the first option by default. Consider a case where, prior to running the tool, the user adds tor or privoxy into `PATH` manually which (generally) is a way to override which executable is used in subsequent steps of a script. It might be nice to use `where` in such a case since I believe is documented to respect PATH.

Additionally, isn't enforcing `/bin/` being in the path an unnecessary restriction? It seems to me that as long as `where` returns a non-empty result for an executable name, we can just put that executable name in `ExecutablePathOverride` and forget about the absolute full path to it (much like what privoxy does today with `PrivoxySettings.ExecutablePathOverride`).

Reply:
It's whereis simply because of linux command whereis.

> Additionally, isn't enforcing `/bin/` being in the path an unnecessary restriction

Actually search in bin directory can be removed because whereis by default search in limited places as PATH variable.
But if user know where he put executable is it much easy for him just set **ExecutablePathOverride** them self, no?

```
       By  default whereis tries to find files from hard-coded paths, which are defined with glob
       patterns.  The command attempts to use the contents  of  $PATH  and  $MANPATH  environment
       variables as default search path.  The easiest way to know what paths are in use is to add
       the -l listing option.  Effects of the -B, -M, and -S are displayed with -l.
```

https://manpages.ubuntu.com/manpages/bionic/man1/whereis.1.html

> we can just put that executable name in ExecutablePathOverride

I think this is can be also target directory when binary will be searched.

Reply:
> But if user know where he put executable is it much easy for him just set ExecutablePathOverride them self, no?

Yes, I agree. But from my play with `whereis` it seems to provide a more complex set of output and behavior than the simple `which` tool which is also available on all distros I've used (perhaps maybe not arm64 distros and that's what I'm missing)? Maybe I was naive in thinking "which" is the canonical (no pun intended) way to find the resolve path of an executable.

Reply:
Seems `which` is better solution here. I now remember why regex here, it's because `whereis` return more that just executable and regex needs to select from multiple matches. `which` return only executable.